packages/@react-stately/collections/docs/collections.mdx

```diff
@@ -0,0 +1,387 @@
+
+```tsx
+import {useListData} from '@react-stately/data';
+
+let list = useListData({
+  initialItems: [
+    {name: 'Aardvark'},
+    {name: 'Kangaroo'},
+    {name: 'Snake'}
+  ],
+  initialSelectedKeys: ['Kangaroo'],
+  getKey: item => item.name
+});
+
+function addAnimal(name) {
+  list.append({name});
+}
+
+<ListBox items={list.items} itemKey="name">
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+### Sections
+
+Sections can be built by returning a &lt;<TypeLink links={docs.links} type={docs.exports.Section} />&gt; instead of
+an &lt;<TypeLink links={docs.links} type={docs.exports.Item} />&gt; item from the top-level item renderer. Sections
+also support an `items` prop with `itemKey` and a renderer function for their children.
+
+```tsx
+let [sections, setSections] = useState([
+  {
+    name: 'People',
+    items: [
+      {name: 'David'},
+      {name: 'Same'},
+      {name: 'Jane'}
+    ]
+  },
+  {
+    name: 'Animals',
+    items: [
+      {name: 'Aardvark'},
+      {name: 'Kangaroo'},
+      {name: 'Snake'}
+    ]
+  }
+]);
+
+<Picker items={sections} itemKey="name">
+  {section =>
+    <Section title={section.name} items={section.items} itemKey="name">
+      {item => <Item>{item.name}</Item>}
+    </Section>
+  }
+</Picker>
+```
+
+When updating nested data, be sure that all parent items change accordingly. Items are immutable,
+so don't use mutating methods like push, or replace a property on a parent item. Instead, copy
+the items that changed as needed. You can do this by hand if you like, but it can be challenging.
+To make it easier, you can use the <TypeLink links={dataDocs.links} type={dataDocs.exports.useTreeData} />
+hook. `useTreeData` will also handle automatically removing items from the selection when they are
+removed from the list. See the [useTreeData docs](useTreeData.html) for more details.
+
+```tsx
+import {useTreeData} from '@react-stately/data';
+
+let tree = useTreeData({
+  initialItems: [
+    {
+      name: 'People',
+      items: [
+        {name: 'David'},
+        {name: 'Sam'},
+        {name: 'Jane'}
+      ]
+    },
+    {
+      name: 'Animals',
+      items: [
+        {name: 'Aardvark'},
+        {name: 'Kangaroo'},
+        {name: 'Snake'}
+      ]
+    }
+  ],
+  getKey: item => item.name,
+  getChildren: item => item.items
+});
+
+function addPerson(name) {
+  tree.append('People', {name});
+}
+
+<ListBox items={tree.items}>
+  {node =>
+    <Section title={node.value.name} items={node.children}>
+      {node => <Item>{node.value.name}</Item>}
+    </Section>
+  }
+</ListBox>
+```
+
+## Heirarchical data
+
+Sections only allow a single level of heirarchy. Some components like trees allow arbitrary heirarchical
+data of any depth to be displayed. This can be accomplished by nesting 
+&lt;<TypeLink links={docs.links} type={docs.exports.Item} />&gt; components, and using the `title` prop 
+to specify the contents of items with children.
+
+```tsx
+<Tree>
```

Comment:
Might need to remove this section or find another component to demo it on if Tree doesn't ship